update_blocklist.nix

```
@@ -40,16 +40,43 @@ in
   # Create an ip set and add each ip to it one by one
 
   # IPv4 and IPv6 regex patterns with CIDR notation support - WARNING: might not be correct for all IPs (e.g. ignore valid ones or accept wrong ones), but seems to work fine
+  # Also supports hosts files, as well as plain lists of domains.
   ipv4_regex="^([0-9]{1,3}\.){3}[0-9]{1,3}(\/[0-9]{1,2})?$"
   ipv6_regex="^([0-9a-fA-F:]+::?[0-9a-fA-F]*)+(\/[0-9]{1,3})?$"
+  domain_regex="^[a-zA-Z0-9.-]*[a-zA-Z][a-zA-Z0-9.-]*$"
+  host_regex="^0\.0\.0\.0 ''${domain_regex:1}"
```

Ah! Just when looking deeper at the line above I noticed one thing: because this is using double quotes, the resulting host_regex does not have `\` in them. Instead it will replace `\.` with `.` when parsing the string, which still matches a literal dot, but is not as strict as we wanted, since it could match `0a0b0c0d` ^^
```suggestion
  host_regex='^0\.0\.0\.0 '"''${domain_regex:1}"
```
or
```suggestion
  host_regex="^0\\.0\\.0\\.0 ''${domain_regex:1}"
```